sdk/src/types/block/address/bech32.rs

```diff
@@ -222,11 +228,19 @@ impl<T: core::borrow::Borrow<Bech32Address>> From<T> for Address {
     }
 }
 
-#[cfg(feature = "serde")]
-crate::string_serde_impl!(Bech32Address);
-
 impl<T: AsRef<str> + Send> ConvertTo<Bech32Address> for T {
     fn convert(self) -> Result<Bech32Address, Error> {
         Bech32Address::try_from_str(self)
     }
 }
+
+/// A valid bech32 encoded address.
+#[derive(Clone, Debug, Eq, PartialEq, Hash, AsRef, Deref, Ord, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Should serde fail if it's an invalid bech32? Probably?

Reply:
Thought that was the point of the type